packages/service-worker/worker/src/driver.ts

```
@@ -177,6 +178,16 @@ export class Driver implements Debuggable, UpdateSource {
       return;
     }
 
+    // Under some circumstances (possibly related to opening Chrome DevTools), requests are made
+    // with `cache: 'only-if-cached'` and `mode: 'no-cors'`. These request will eventually fail,
+    // because `only-if-cached` is only allowed to be used with `mode: 'same-origin'`.
+    // This is likely a bug in Chrome DevTools. Avoid handling such requests.
+    // (See also https://github.com/angular/angular/issues/22362.)
+    // TODO(gkalpak): Remove once no longer necessary (i.e. fixed in Chrome DevTools).
+    if ((req.cache as string) === 'only-if-cached' && req.mode !== 'same-origin') {
+      return;
```

Let's `console.error` the first time this happens so it's not a silent failure.

---

Why a failure? It is just that the SW won't handle this request, so it will pass through (and I assume the browser will do the right thing).
(Interestingly, no error is logged when the SW doesn't handle such requests.)

---

I think if someone is looking in DevTools, expecting to see their request answered `(from ServiceWorker)`, that having a message locally there saying "hey, the SW didn't handle this request because DevTools sent it in an invalid way" is beneficial.

---

As discussed "offline", this is not a request made by the app/user, but a background request made by DevTools (apparently). So, we'll `debugger.log()` the first time we see it.

I have pushed another commit. PTAL